opentaskspal/src/main/java/org/dmfs/opentaskspal/tasks/TimeData.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2017 dmfs GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dmfs.opentaskspal.tasks;
+
+import android.content.ContentProviderOperation;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import org.dmfs.android.contentpal.RowData;
+import org.dmfs.android.contentpal.TransactionContext;
+import org.dmfs.rfc5545.DateTime;
+import org.dmfs.tasks.contract.TaskContract;
+
+
+/**
+ * {@link RowData} for adding start, start+due, or start+duration times.
+ *
+ * @author Gabor Keszthelyi
+ */
+// TODO Unit test
+public final class TimeData implements RowData<TaskContract.Tasks>
+{
+    @NonNull
+    private final DateTime mStart;
+    @Nullable
+    private final DateTime mDue;
```

Comment:
Maybe make them (`mDue` and `mDuration`) `Optional` instead.

Reply:
Right, I've changed it. We still use `@Nullable` at the ctor params? That's what I am not sure of, I don't see the end usage - is it going to be part of an 'automatic' syncronization mapping where it is good if 'null' can be passed on, or is the 'normal' usage, like how I would use the class from UI code is what we should aim for? In this latter case, I would put `Optional` in the constructor as well.

Reply:
No, we should not use null at all here. 

it should look like this:
```java
  private TimeData(@NonNull DateTime start, @NonNull Optional<DateTime> due, @NonNull Optional<Duration> duration)
    {
        mStart = start;
        mDue = due;
        mDuration = duration;
    }


    public TimeData(@NonNull DateTime start, @NonNull DateTime due)
    {
        this(start, new Present<>(due), absent());
    }

…
```

Reply:
Ok, updated.